src/txadmission.cpp

```diff
@@ -327,22 +349,27 @@ void CommitTxToMempool()
         // from re-requests.
         LOG(MEMPOOL, "popping txdeferQ, size %d\n", txDeferQ.size());
         // this could be a lot more efficient
-        while (!txDeferQ.empty())
+        uint64_t count = 0;
+        uint64_t maxmove = max(aveCommitBatchSize * 2, minCommitBatchSize);
+        while ((!txDeferQ.empty()) && (count < maxmove))
```

Comment:
nit:  ave => avg   , throws me for a loop when I see "ave"

https://writingexplained.org/english-abbreviations/average

Reply:
great!  "They" made me use AVE for the leaky bucket stuff way back...